portal-api/src/v1/gef/services/clone-deal.service.test.ts

```diff
@@ -0,0 +1,245 @@
+          auditDetails: mockAuditDetails,
+        }),
+      ).rejects.toThrow(new DealNotFoundError(existingDealId));
+    });
+
+    it('gets existing deal', async () => {
+      // Act
+      await cloneDealToLatestVersion({
+        dealId: existingDealId,
+        bankInternalRefName,
+        additionalRefName,
+        maker,
+        userId: maker._id,
+        bank: mockBank,
+        auditDetails: mockAuditDetails,
+      });
+
+      // Assert
+      expect(findOneMock).toHaveBeenCalledTimes(1);
+      expect(findOneMock).toHaveBeenCalledWith({ _id: { $eq: new ObjectId(existingDealId) }, 'bank.id': { $eq: mockBank.id } });
+    });
+
+    it('inserts a cloned deal', async () => {
+      // Act
+      await cloneDealToLatestVersion({
+        dealId: existingDealId,
+        bankInternalRefName,
+        additionalRefName,
+        maker,
+        userId: maker._id,
+        bank: mockBank,
+        auditDetails: mockAuditDetails,
+      });
+
+      // Assert
+      expect(insertOneMock).toHaveBeenCalledTimes(1);
+      expect(insertOneMock).toHaveBeenCalledWith({
+        ...generateCloneDeal({
+          dealId: existingDealId,
+          bankInternalRefName,
+          additionalRefName,
+          maker,
+          userId: maker._id,
+          bank: mockBank,
+          auditDetails: mockAuditDetails,
+          existingDeal,
+          latestEligibilityCriteria: mockLatestEligibilityCriteria,
+        }),
+        _id: expect.any(ObjectId) as ObjectId,
+      });
+    });
+  });
+
+  describe('generateCloneDeal', () => {
+    it('returns the expected clone deal', () => {
+      // Act
+      const cloneDeal = generateCloneDeal({
+        dealId: existingDealId,
+        bankInternalRefName,
+        additionalRefName,
+        maker,
+        userId: maker._id,
+        bank: mockBank,
+        auditDetails: mockAuditDetails,
+        existingDeal,
+        latestEligibilityCriteria: mockLatestEligibilityCriteria,
+      });
+
+      // Assert
+      expect(cloneDeal).toEqual({
+        _id: expect.any(ObjectId) as ObjectId,
+        dealType: existingDeal.dealType,
+        version: getCurrentGefDealVersion(),
+        createdAt: Date.now(),
+        updatedAt: Date.now(),
+        facilitiesUpdated: Date.now(),
+        eligibility: {
+          ...mockLatestEligibilityCriteria,
+          updatedAt: Date.now(),
+        },
+        status: DEAL_STATUS.DRAFT,
+        submissionType: null,
+        submissionDate: null,
+        submissionCount: 0,
+        bankInternalRefName,
+        additionalRefName,
+        maker,
+        bank: mockBank,
+        ukefDealId: null,
+        checkerId: null,
+        editedBy: [maker._id],
+        exporter: cloneExporter(existingDeal.exporter),
+        portalActivities: [],
+        supportingInformation: {},
+        clonedDealId: existingDealId,
+        auditRecord: generateAuditDatabaseRecordFromAuditDetails(mockAuditDetails),
+      });
+    });
+  });
+});
```

Comment:
There's lots going on in the service, can we split into new directories? E.g

```ts
clone-deal.service/index.ts
clone-deal.service/generate-cloned-deal.ts
```

Concerned if either of the functions/tests grow